cli/src/pcluster/aws/ec2.py

```diff
@@ -221,25 +228,39 @@ def get_official_image_id(self, os, architecture, filters=None):
             raise AWSClientError(function_name="describe_images", message="Cannot find official ParallelCluster AMI")
         return max(images, key=lambda image: image["CreationDate"]).get("ImageId")
 
+    def get_official_images(self, version=None, os=None, architecture=None):
+        """Get the list of official images, optionally filtered by os and architecture."""
+        try:
+            owners = ["amazon"]
+            version = version or "*"
+            os = OS_TO_IMAGE_NAME_PART_MAP.get(os, "") if os else "*"
+            architecture = architecture or "*"
+            name = f"aws-parallelcluster-{version}-{os}-{architecture}*"
+            filters = [{"Name": "name", "Values": [name]}]
+            return self.describe_images(ami_ids=[], owners=owners, filters=filters)
+        except ImageNotFoundError:
+            return []
+
     @AWSExceptionHandler.handle_client_exception
     @Cache.cached
     def get_eip_allocation_id(self, eip):
         """Retrieve the allocation id of an Elastic IP."""
         return self._client.describe_addresses(PublicIps=[eip])["Addresses"][0]["AllocationId"]
 
+    @staticmethod
+    def extract_os_from_official_image_name(name):
```

Comment:
not sure I'd leave this in the Ec2Client class, what do you think?

Reply:
I don't really like it either from a "semantic" point of view, as this is not an EC2 call, however my reasoning is that I wanted to keep it close to where the "pattern" of the official ami name was defined, so that if we decided to change the naming (i.e. from `aws-parallelcluster-...` to something else) we would have to make the change is a single place.